File: 445-add-two-numbers-ii/add-two-numbers-ii.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
import java.math.*;
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        // List<Integer> list1=new ArrayList<>();
        // List<Integer> list2=new ArrayList<>();
        String str1="",str2="";
        ListNode temp1=l1;
        ListNode temp2=l2;
        while(temp1!=null){
            str1+=String.valueOf(temp1.val);
            temp1=temp1.next;
        }
        while(temp2!=null){
            str2+=String.valueOf(temp2.val);
            temp2=temp2.next;
        }
        String res=String.valueOf(new BigInteger(str1)
                                    .add(new BigInteger(str2)));
        ListNode dummy=new ListNode(0);
        ListNode t=dummy;
        for(char c:res.toCharArray()){
            t.next=new ListNode(c-'0');
            t=t.next;
        }
        return dummy.next;
        
    }
}

File: 445-add-two-numbers-ii/add-two-numbers-ii.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        str1=str2=""
        temp1,temp2=l1,l2
        while temp1:
            str1+=str(temp1.val)
            temp1=temp1.next
        while temp2:
            str2+=str(temp2.val)
            temp2=temp2.next
        res=str(int(str1)+int(str2)) 
        dummy=ListNode(0)
        t=dummy
        for x in res:
            t.next=ListNode(int(x))
            t=t.next
        return dummy.next
